pkg/expr/expr_test.go

```diff
@@ -63,6 +65,81 @@ func TestGoodParse(t *testing.T) {
 	}
 }
 
+func TestExtractMatches(t *testing.T) {
+	for _, tc := range []struct {
+		desc string
+		src  string
+		m    map[string]interface{}
+	}{
+		{
+			desc: "no ANDS",
+			src:  `a || b || "c" || ( a && b )`,
+			m:    map[string]interface{}{},
+		},
+		{
+			desc: "EQ check with function",
+			src:  `substring(a, 5) == "abc"`,
+			m:    map[string]interface{}{},
+		},
+		{
+			desc: "single EQ check",
+			src:  `origin.host == "9.0.10.1"`,
+			m: map[string]interface{}{
+				"origin.host": "9.0.10.1",
+			},
+		},
```

Comment:
maybe nicer to have a case of the opposite direction, i.e. `"9.0.10.1" == origin.host`

Reply:
done.